ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -6680,3 +6701,89 @@ static int set_default_authz_handler(ldmsd_req_ctxt_t reqc)
 
 	return 0;
 }
+
+extern struct option long_opts[];
+extern char *short_opts;
+extern int ldmsd_process_cmd_line_arg(char opt, char *value);
+static int __cmdline_options(ldmsd_req_ctxt_t reqc, int argc, char *argv[])
+{
+	int opt, opt_idx;
+	reqc->errcode = 0;
+	opterr = 0;
+	optind = 0;
+	while (0 < (opt = getopt_long(argc, argv,
+				  short_opts, long_opts,
+				  &opt_idx))) {
+		switch (opt) {
+		case 'x':
+			reqc->errcode = EINVAL;
+			snprintf(reqc->line_buf, reqc->line_len,
+					"The 'option' command does not support 'x' or 'xprt'. "
+					"Use the 'listen' command instead.");
+			return reqc->errcode;
+		case 'F':
+		case 'u':
+		case 'V':
+			reqc->errcode = EINVAL;
+			snprintf(reqc->line_buf, reqc->line_len,
+					"The option '%s' must be given at the command line.",
+					argv[optind-1]);
+			return reqc->errcode;
+		default:
+			reqc->errcode = ldmsd_process_cmd_line_arg(opt, optarg);
+			if (reqc->errcode == ENOENT) {
+				snprintf(reqc->line_buf, reqc->line_len,
+					"Unknown cmd-line option or it must be "
+					"given at the command line: %s\n", argv[optind-1]);
+				return reqc->errcode;
+			}
+			break;
+		}
+	}
+	return 0;
+}
+
+static int cmd_line_arg_set_handler(ldmsd_req_ctxt_t reqc)
+{
+	char *s, *token, *ptr1, *lval;
+	int rc = 0;
+	int argc = 1;
+	int i;
+	char *argv[11];
```

Comment:
What is the rationale for hardcoding this as "11"? Probably needs a code comment at the very least.